WcaOnRails/app/helpers/competitions_helper.rb

```diff
@@ -2,26 +2,39 @@
 
 module CompetitionsHelper
   def competition_message_for_user(competition, user)
-    messages = []
-    registration = competition.registrations.find_by_user_id(user.id)
-    if competition.cancelled?
-      messages << t('competitions.messages.cancelled')
-    elsif registration
-      # Display a tooltip on a user's bookmarked competition based on their registration status.
-      if registration.accepted?
-        messages << t('competitions.messages.tooltip_registered')
-      elsif registration.deleted?
-        messages << t('competitions.messages.tooltip_deleted')
-      else # If not delted or accepted, assume user is on the waiting list
-        messages << t('competitions.messages.tooltip_waiting_list')
-      end
+    # Generates a list of messages, which will be combined and displayed in a tooltip to the user in their bookmarked
+    # competitions list when they hover over a competition.
+    # Message indicates the state of the competition, and the state of the user's registration.
+
+    messages = [] # Messages to be combined
+
+    if competition.cancelled? # If the competition is cancelled, just return the cancellation message to the user.
```

Comment:
Nice, I like the changes in general, but I think it might be a little bit over-commented in places. This comment, for example, is literally restating the code, and isn't really significantly clearer. Plus if this gets changed in the future, both the code and the comment needing updating, and it's possible the comment will be missed, leading to confusion down the line.
Same with the comment above the `join`.

Reply:
Great points, especially about needing to change two things instead of 1, and using things like variable names to communicate without the need for comments

I've opted to leave the comment next to `competition.cancelled?`, to explain the logic/intention of returning immediately there. It might just be my baby brain, but it feels like it makes it easier to understand :P